sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOrcScan.scala

```diff
@@ -280,6 +280,49 @@ object GpuOrcScan extends Arm {
   def testLongMultiplicationOverflow(a: Long, b: Long) = {
     Math.multiplyExact(a, b)
   }
+
+
+  /**
+   * Convert the integer vector into timestamp(microseconds) vector.
+   * @param col The integer columnar vector.
+   * @param colType Specific integer type, it should be BOOL/INT8/INT16/INT32/INT64.
+   * @param timeUnit It should be one of {DType.TIMESTAMP_SECONDS, DType.TIMESTAMP_MILLISECONDS}.
+   *                 If timeUnit == SECONDS, then we consider the integers as seconds.
+   *                 If timeUnit == MILLISECONDS, then we consider the integers as milliseconds.
+   *                 This parameter is determined by the shims.
+   * @return A timestamp vector.
+   */
+  def castIntegersToTimestamp(col: ColumnView, colType: DType,
+                              timeUnit: DType): ColumnVector = {
+    assert(colType == DType.BOOL8 || colType == DType.INT8 || colType == DType.INT16
+      || colType == DType.INT32 || colType == DType.INT64)
+    assert(timeUnit == DType.TIMESTAMP_SECONDS || timeUnit == DType.TIMESTAMP_MILLISECONDS)
+
+    colType match {
+      case DType.BOOL8 | DType.INT8 | DType.INT16 | DType.INT32 =>
+        // cuDF requires casting to Long first, then we can cast Long to Timestamp(in microseconds)
+        withResource(col.castTo(DType.INT64)) { longs =>
+          // bitCastTo will re-interpret the long values as 'timeUnit', and it will zero-copy cast
+          // between types with the same underlying length.
+          longs.bitCastTo(timeUnit).castTo(DType.TIMESTAMP_MICROSECONDS)
```

Comment:
```suggestion
          withResource(longs.bitCastTo(timeUnit)) { timeView =>
            timeView.castTo(DType.TIMESTAMP_MICROSECONDS)
          }
```
The ColumnView from `longs.bitCastTo(timeUnit)` also needs to be closed.

Reply:
I did such thing in the previous code. But it will throw some exceptions like "Close Column Vector too many time....".

The description of `bitCastTo` is:

```
public ColumnView bitCastTo(DType type)
Zero-copy cast between types with the same underlying length. Similar to bit_cast in C++. This will take the underlying data and create new metadata so it is interpreted as a new type. Not all types are supported the width of the types must match.
```

it returns an ColumnView (not CoolumnVector), so I think there is no need to close it?

Reply:
Fixed.